TheWarWithin/MageArcane.lua

```diff
@@ -1269,27 +1288,27 @@ spec:RegisterAbilities( {
                 if buff.presence_of_mind.down then setCooldown( "presence_of_mind", 60 ) end
             end
 
-            removeStack( "concentration" )
+            removeBuff( "concentration" )
             removeBuff( "leydrinker" )
-            
-            if buff.next_blast_spheres.stacks == 6 then
-                removeBuff ( "next_blast_spheres" )
-                addStack ( "spellfire_spheres" )
-                applyBuff ( "burden_of_power" )
-            elseif buff.next_blast_spheres.up then
-                addStack ( "next_blast_spheres" )
-            else
-                applyBuff ( "next_blast_spheres" )
+
+            if talent.spellfire_spheres.enabled then
+                if buff.next_blast_spheres.stacks == 5 then
```

Comment:
If this triggers at 5, then this check should be `== 4` since we're not applying the new stack before checking.

Reply:
while your character has 5 stacks, the cast makes buff go bye bye and triggers the other stuff. I incorrectly had `== 6` before

Reply:
Does it happen as soon as you reach 5 stacks, or does it happen when you cast and you already have 5 stacks?

Reply:
You've got 5 stacks and press your button. When cast finishes, it goes away and becomes a new buff.

If counter buff doesn't exist, the cast starts it over at 1 again.